charts/orchestrator/templates/rhdh-operator.yaml

```diff
@@ -0,0 +1,215 @@
+{{- if .Values.rhdhOperator.enabled }}
+{{- $unmanagedNamespaceExists := include "unmanaged-resource-exists" (list "v1" "Namespace" "" .Values.rhdhOperator.subscription.namespace .Release.Name) }}
+{{- if and (eq $unmanagedNamespaceExists "false") .Values.rhdhOperator.enabled }}
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: {{ .Values.rhdhOperator.subscription.namespace }}
+---
+apiVersion: operators.coreos.com/v1
+kind: OperatorGroup
+metadata:
+  name: rhdh-operator-group
+  namespace: {{ .Values.rhdhOperator.subscription.namespace }}
+spec: {}
+---
+{{- end }}
+{{- if ne .Values.rhdhOperator.subscription.sourceImage "" }}
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: {{ .Values.rhdhOperator.subscription.source }}
+  namespace: openshift-marketplace
+spec:
+  displayName: RHDH Operator (Development)
+  image: {{ .Values.rhdhOperator.subscription.sourceImage }}
+  sourceType: grpc
+---
+{{- end }}
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: rhdh-operator
+  namespace: {{ .Values.rhdhOperator.subscription.namespace }}
+spec:
```

Comment:
If you want to simplify the template and the documentation and, at the same time, provide more flexibility so that new sub-fields can be added if needed, you can just have a single structured JSON object representing the full configuration and then install it as:
```yaml
spec:
{{ toYaml .Values.subscription.spec | indent 2 }}
```
Then you can provide a default in your values.yaml with a sample data. (in the above example should be under `rhdhOperator.subscription.spec`, of course you can name it as you prefer